macroid-core/src/main/scala/macroid/MultiEventTweaks.scala

```diff
@@ -0,0 +1,146 @@
+  }
+
+  object UnitFuncOn extends Dynamic {
+    def applyDynamic[W <: View](event: String)(handler: Any): Tweak[W] = macro onUnitFuncImpl[W]
+  }
+
+  object MultiOn extends Dynamic {
+    def applyDynamic[W <: View](event: String)(handlers: (String => Any)*): Tweak[W] = macro onMultipleFuncImpl[W]
+  }
+}
+
+object MultiEventTweaks extends MultiEventTweaks
+
+private object MultiEventTweakMacros {
+
+  import MultiEventTweaks._
+
+  def onUnitImpl[W <: View : c.WeakTypeTag](c: MacroContext)(event: c.Expr[String])(handler: c.Expr[Any]) = {
+    import c.universe._
+
+    val Expr(Literal(Constant(e: String))) = event
+
+    onMultipleFuncImpl(c)(event)(c.Expr[UnitHandler => Any](q"(${newTermName(s"on${e.capitalize}")}: UnitHandler) => $handler"))
+  }
+
+  def onUnitFuncImpl[W <: View : c.WeakTypeTag](c: MacroContext)(event: c.Expr[String])(handler: c.Expr[Any]) = {
+    import c.universe._
+
+    val Expr(Literal(Constant(e: String))) = event
+
+    onMultipleFuncImpl(c)(event)(c.Expr[FuncHandler => Any](q"(${newTermName(s"on${e.capitalize}")}: FuncHandler) => $handler"))
+  }
+
+  def onMultipleFuncImpl[W <: View : c.WeakTypeTag](c: MacroContext)(event: c.Expr[String])(handlers: (c.Expr[String => Any])*) = {
+    import c.universe._
+
+    val tpe = weakTypeOf[W]
+    val (_, setter) = findSetter(c)(event, tpe)
+
+    val (listener, handlerOnMethodTupleSequence) = onBaseMultipleHandlers(c)(setter, handlers)
+
+    val overrides = handlerOnMethodTupleSequence map { case (handler, on, handlerType) =>
+      getOverride(c)(on, c.Expr(c.resetLocalAttrs(handler.tree.children.last)), handlerType)
+    }
+
+    c.Expr[Tweak[View]](q"_root_.macroid.Tweak[$tpe] { x ⇒ x.$setter(new $listener { ..$overrides })}")
+  }
+
+  def findSetter(c: MacroContext)(event: c.Expr[String], tp: c.Type) = {
+    import c.universe._
+
+    val Expr(Literal(Constant(eventName: String))) = event
+    val setter = scala.util.Try {
+      val s = tp.member(newTermName(s"setOn${eventName.capitalize}Listener")).asMethod
+      assert(s != NoSymbol)
+      s
+    } getOrElse {
+      c.abort(c.enclosingPosition, s"Could not find method setOn${eventName.capitalize}Listener in $tp. " +
+          s"You may need to provide the type argument explicitly")
+    }
+    (eventName, setter)
+  }
+
+  def onBaseMultipleHandlers(c: MacroContext)(setter: c.universe.MethodSymbol, handlers: Seq[c.Expr[String => Any]]) = {
+    import c.universe._
+
+    val listener = setter.paramss(0)(0).typeSignature
+    val handlerOnMethodTupleSequence = handlers map { h =>
+      val (eventName, typ) = h.tree.children.head match {
+        case ValDef(_, name, tpt, _) => (name.toString, tpt)
+      }
+
+      val handlerType = typ.toString() match {
+        case s: String if s.endsWith("UnitHandler") => HUnit
+        case s: String if s.endsWith("FuncHandler") => HUnitFunc
+      }
+
+      val on = scala.util.Try {
```

Comment:
I think this also duplicates `EventTweakMacros.onBase` — would be nice to reuse the same method.